CONTRIBUTING.md

```diff
@@ -0,0 +1,187 @@
+# CONTRIBUTING to Flex
+
+## Introduction
+
+Thanks for your interest in contributing to Flex! Flex's [issue tracker](https://github.com/westes/flex/issues) is on GitHub. That's the best place to find a task that needs attention. It's also the best place to check whether a problem you've found is known to the community.
+
+## Baseline Build Environment
+
+The maintenance baseline build environment is Ubuntu 20.04 LTS with at least the following packages installed:
+- bash
+- gcc
+- m4
+- autoconf
+- automake
+- make
+- bison
+- gettext
+- autopoint
+- help2man
+- tar
+- gzip
+- lzip
+- texinfo
+- texlive
+- indent
+- flex (flex bootstraps itself from its previous version)
+
+Configuring your Flex development environment to match the maintenance baseline will help you collaborate and review contributions with the other developers working on Flex.
+
+## Branching Convention
+
+Flex source code is maintained in Git at GitHub. Each Flex developer forks their own copy of the main repository at [westes/flex](https://github.com/westes/flex). Flex development occurs in feature branches of the forks. PRs will eventually be submitted from the feature branches of the forks to westes/flex. 
+
+Flex development forks are not required to reside on GitHub, merely on a publicly-accessible Git server. However:
+- GitHub's pull request (PR) process only works with repositories hosted on GitHub;
+- Flex's automated testing pipeline depends on GitHub's Actions infrastructure and may not function on other Git servers.
+
+NB: If you cannot use GitHub, another developer can help you import your changes to a GitHub name space using multiple remotes. The details are beyond the scope of this manual.
+
+## Maintaining the Test Suite
+
+### Introduction to Flex Tests
+Flex includes an extensive test suite in the `tests/` directory. It includes functional tests as well as regression tests. If you add a feature to Flex, develop corresponding tests for the correct behaviors. Whenever you develop a bug fix, develop a corresponding test for the incorrect behavior to expose future regressions.
+
+The test suite is controlled by `tests/Makefile.am`. Each test consists of a Flex scanner specification, `<category>.l`, that exercises the behavior under test as specifically as possible and an input text file, `<category>.txt`, containing strings for the test scanner to recognize. Flex tests return 0 (zero) on success and 1 (or non-zero) on failure. Test return values are generated by a main() function defined in the test scanner.
```

Comment:
The test exit code specifications are those of automake. E.g. 77 for "skip." There is no requirement that main() do anything in particular, so long as the results are as expected by automake.